arcwise/agent/execute_sql.py

```diff
@@ -147,21 +153,13 @@ async def execute_sql(
 async def _execute_sqlite(sql: str, sql_context: SQLContext) -> list[list[str]]:
     process = None
     try:
-        process = await asyncio.create_subprocess_exec(
-            "sqlite3",
-            "-csv",
-            "-header",
-            sql_context.db_url,
-            sql,
-            stdin=asyncio.subprocess.PIPE,
-            stdout=asyncio.subprocess.PIPE,
-            stderr=asyncio.subprocess.PIPE,
-        )
-        stdout, stderr = await process.communicate()
-        if process.returncode != 0:
-            raise RuntimeError(f"SQLite error: {stderr.decode()}")
-
-        return list(csv.reader(StringIO(stdout.decode()), delimiter=","))
+        conn = sqlite3.connect(sql_context.db_url)
+        conn.row_factory = sqlite3.Row
+        cursor = conn.cursor()
+        cursor.execute(sql)
+        columns = [col[0] for col in cursor.description]
+        rows = cursor.fetchall()
+        return [columns] + [[row[col] for col in columns] for row in rows]
```

Comment:
Not a critical change, but thought this could help keep our answers and their grader's consistent

Reply:
Unfortunately cannot make this change as the vllm base image we’re using in the docker file is stuck at Python 3.10 :(

Reply:
Oh dang, actually... I just recalled that there was more to this change: this was actually also related to checking the output schema & types.  By creating the subprocess, everything comes back as strings.

Reply:
I see. Would it be sufficient just to check the number of columns? Should be reasonable to inspect the report message logs to see if it actually confuses the types or if it only adds extra columns for example.

If the type checking is critical then we can still use the static extract_sql_references code to do that.

Reply:
I had seen at least one instance where it had the right answer, but it had the columns flipped, e.g. INTEGER, TEXT when it was asking for TEXT, INTEGER, which prompted me to think that should be squashable.